### DISCUSSION TITLE
how to install blue blocker manually if firefox rejected it?

### DISCUSSION THREAD

**Discussion by rueqen** [kheina-com/Blue-Blocker#302] [Q&A] [answered]
i heard firefox rejected the new version and i'm wondering how do i install it myself without relying on the add-on store.

**Comment by kheina** [comment, marked as answer]
you can either follow the directions here https://github.com/kheina-com/Blue-Blocker?tab=readme-ov-file#firefox (but just select the 0.4.2 .zip file) which will unfortunately remove the add-on every time you restart Firefox. or, you can Google how to disable extension signature verification and load the .zip from the releases tab. this is NOT recommended as it compromises part of Firefox's security steps to load unverified add-ons. I will not provide explicit instructions on how to do this.

I'm very sorry, I wish there was a better way while Firefox's review team sits on their hands

**Reply by Baramanga** [reply]
This maybe a dumb question but where do I run those commands? I'm not super familiar with firefox command environments. Are these Linux commands? Is there a step I need to set up so that whatever that environment is can link to Firefox?

Yeah I don't think I know how to set up Node.js so that it can use your provided commands. It told me:
`npm should be run outside of the Node.js REPL, in your normal shell.`

**Reply by kheina** [reply]
Skip 1 and 2 and just do 3 and 4 but use the .zip instead of the manifest

**Reply by actualb3n** [reply]
hi, I'm getting this error message when using the zip file. anything i can do about it?

> File F:\Users\Ben\Downloads\Blue-Blocker-0.4.11-workflow.1.zip does not contain a valid manifest

**Reply by rougetimelord** [reply]
> hi, I'm getting this error message when using the zip file. anything i can do about it?
> 
> > File F:\Users\Ben\Downloads\Blue-Blocker-0.4.11-workflow.1.zip does not contain a valid manifest

I think you're trying to load the source code zip as an extension, check the v0.4.12 release for a zip that's actually built

**Reply by actualb3n** [reply]
> > hi, I'm getting this error message when using the zip file. anything i can do about it?
> > > File F:\Users\Ben\Downloads\Blue-Blocker-0.4.11-workflow.1.zip does not contain a valid manifest
> 
> I think you're trying to load the source code zip as an extension, check the v0.4.12 release for a zip that's actually built

Nice, that works a lot better. thanks a lot!
